src/Compiler/Compiler.cpp

```diff
@@ -1013,6 +1017,51 @@ bool Compiler::RegisterCommands(TclInterpreter* interp, bool batchMode) {
     interp->registerCmd("detailed_placement", placement, this, 0);
     interp->registerCmd("place", placement, this, 0);
 
+    auto pin_loc_assign_method = [](void* clientData, Tcl_Interp* interp,
+                                    int argc, const char* argv[]) -> int {
+      Compiler* compiler = (Compiler*)clientData;
+      auto setPlaceOption = [compiler](const std::string& arg) {
+        if (arg == "random") {
+          flag = false;
+          compiler->PinAssignOpts(Compiler::PinAssignOpt::Random);
+          compiler->Message("Pin Method set -> [Random]");
+        } else if (arg == "in_define_order") {
+          flag = true;
+          compiler->PinAssignOpts(Compiler::PinAssignOpt::In_Define_Order);
+          compiler->Message("Pin Method set -> [In Define Order]'");
+        } else {
+          compiler->ErrorMessage("Unknown Placement Option: " + arg);
+        }
+      };
+
+      // If we received a tcl argument
+      if (argc > 1 && argc < 4) {
+        setPlaceOption(argv[argc - 1]);
+      } else {
+        compiler->ErrorMessage(
```

Comment:
here is the problem - we show error but something still set despite the error, this is logically wrong. Nothing should be done here.